src/test/java/suite/operator/TestData.java

```diff
@@ -27,6 +39,46 @@ public static Set<Arguments> baseArgs() {
         return args;
     }
 
+    public static Stream<Arguments> sharingArgs() {
+        final var privateOperator = named("PRIVATE OPERATOR", getOperator(Operator.Type.PRIVATE).orElse(null));
+        final var publicOperator = named("PUBLIC OPERATOR", getOperator(Operator.Type.PUBLIC).orElse(null));
+
+        final var client = named("CLIENT", CLIENT);
+        final var sharingRecipient = named("SHARING RECIPIENT", SHARING_RECIPIENT);
+        final var nonSharingRecipient = named("NON-SHARING RECIPIENT", NON_SHARING_RECIPIENT);
+
+        return Stream.of(
+                // CLIENT shares with CLIENT.
+                Arguments.of(client, privateOperator, client, privateOperator, DecryptionStatus.SUCCESS),
+                Arguments.of(client, privateOperator, client, publicOperator, DecryptionStatus.SUCCESS),
+                Arguments.of(client, publicOperator, client, privateOperator, DecryptionStatus.SUCCESS),
+                Arguments.of(client, publicOperator, client, publicOperator, DecryptionStatus.SUCCESS),
+
+                // CLIENT shares with SHARING RECIPIENT.
```

Comment:
maybe add a comment on where (in the other repo i presume) the sharing/non-sharing recipients are defined so that we can confirm there is/isn't a sharing relationship defined

Reply:
It's a bit complicated.

Our test data is defined in [uid2-admin](https://github.com/IABTechLab/uid2-admin/tree/main/src/main/resources/localstack/s3/core).

The environment variables that define the sharing/non-sharing recipients are in [uid2-shared-actions](https://github.com/IABTechLab/uid2-shared-actions) for the pipeline, and in the private repo `uid2-dev-workspace` when running the tests locally.

Reply:
it would be good if you document what you said above as comments so people would know where to look :)

Reply:
[Added a comment](https://github.com/IABTechLab/uid2-e2e/pull/13/files#diff-6836e32e3689865542e898ee347af01b08ba151a66c31283268ce6e3961eff8dR68-R71)